### ISSUE TITLE
Babel error when running react-native-git-upgrade

### ISSUE THREAD

**Issue by Nickersoft** [facebook/react-native#18886]
- [X] I have reviewed the [documentation](https://facebook.github.io/react-native)
- [X] I have searched [existing issues](https://github.com/facebook/react-native/issues)
- [x] I am using the [latest React Native version](https://github.com/facebook/react-native/releases)

## Environment

Environment:
  OS:  macOS High Sierra 10.13.4
  Node:  9.11.1
  Yarn:  1.5.1
  npm:  5.6.0
  Watchman:  4.9.0
  Xcode:  Xcode 9.3 Build version 9E145
  Android Studio:  2.3 AI-162.4069837

Packages: (wanted => installed)
  react: ^16.3.1 => 16.3.1
  react-native: 0.54.2 => 0.54.2

## Steps to Reproduce

I imagine this error occurs with all projects that utilize a 7.x beta version of Babel
 
## Expected Behavior

I'm trying to upgrade from React Native 0.53.x to 0.55 via `react-native-git-upgrade`

## Actual Behavior

I'm receiving the following error directly after the "Generate old version template" step:

```
Error: Requires Babel "^7.0.0-0", but was loaded with "6.26.0". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. 
Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel. (While processing preset: "/Users/nickersoft/rnapp/node_modules/@babel/preset-flow/lib/index.js")
```

I'm currently using `@babel/core` (beta.44) for my project. My hunch is that `react-native-git-upgrade` is loading Babel via its own 6.x dependencies and calling `babel-register`, but once it hits the plugins and code I'm using it errors, seeing those expect 7.x. I've updated my code to use `@babel/core` and would prefer to not downgrade, but I'm curious if that's the only option. I tried the beta release of `react-native-git-upgrade`, thinking it might use Babel beta, but it doesn't. Any help would be appreciated. Thanks.

**Comment by Nickersoft** [comment]
@react-native-bot I would, but this issue revolves around upgrading to v0.55... also, why did I respond to a bot?

**Comment by hramos** [comment]
The next release should use Babel 7 - the work has already begun on master - hopefully that should help with the upgrade.

**Comment by Strate** [comment]
Still actual while trying upgrade from 0.56 to 0.57. react-native-git-upgrade still uses babel@6 @hramos

**Comment by hramos** [comment]
`react-native-git-upgrade` is known to not work as expected even in 0.57. This is part of the cli is community supported, feel free to send a PR if you're interested in helping out here.

**Comment by jay1337** [comment]
This commit solves this issue: https://github.com/facebook/react-native/commit/b4a8630ea59bbc4a2b384c6e88dc5fd82ee5ee5a

**Comment by hramos** [comment]
Fixed in 0.58.
